python/pyspark/ml/tests.py

```diff
@@ -594,17 +594,21 @@ def test_fit_minimize_metric(self):
         iee = InducedErrorEstimator()
         evaluator = RegressionEvaluator(metricName="rmse")
 
-        grid = (ParamGridBuilder()
-                .addGrid(iee.inducedError, [100.0, 0.0, 10000.0])
-                .build())
+        grid = ParamGridBuilder() \
+            .addGrid(iee.inducedError, [100.0, 0.0, 10000.0]) \
+            .build()
         tvs = TrainValidationSplit(estimator=iee, estimatorParamMaps=grid, evaluator=evaluator)
         tvsModel = tvs.fit(dataset)
         bestModel = tvsModel.bestModel
         bestModelMetric = evaluator.evaluate(bestModel.transform(dataset))
+        validationMetrics = tvsModel.validationMetrics
 
         self.assertEqual(0.0, bestModel.getOrDefault('inducedError'),
                          "Best model should have zero induced error")
         self.assertEqual(0.0, bestModelMetric, "Best model has RMSE of 0")
+        self.assertEqual(len(grid), len(validationMetrics),
+                         "validationMetrics has the same size of grid parameter")
+        self.assertEqual(bestModelMetric, min(validationMetrics))
```

Comment:
With @MLnick's previous comment in mind, my preference would be to instead check:

`self.assertEqual(0.0, min(validationMetrics))`

That way we don't put in a check that is wrong in general, but correct here. I don't feel too strongly about it, but since it's an easy fix and we might save some future developer a bit of confusion then I think it's a good idea. Same in the maximize version.